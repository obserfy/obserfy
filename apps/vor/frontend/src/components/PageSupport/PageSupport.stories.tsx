import { FC } from "react"
import PageSupport from "./PageSupport"

export default {
  title: "Core/PageSupport",
  component: PageSupport,
  parameters: {
    componentSubtitle: "Just a simple PageSupport",
  },
}

export const Basic: FC = () => <PageSupport />
