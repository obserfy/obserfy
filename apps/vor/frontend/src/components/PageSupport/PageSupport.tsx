import { FC, useEffect } from "react"
import { Box, Card, Flex, Button } from "theme-ui"
import { Trans } from "@lingui/macro"
import { useGetUserProfile } from "../../hooks/api/useGetUserProfile"
import { loadCanny } from "../../canny"
import Typography from "../Typography/Typography"
import Icon from "../Icon/Icon"
import { ReactComponent as NextIcon } from "../../icons/next-arrow.svg"
import { ReactComponent as MailIcon } from "../../icons/mail.svg"

export const PageSupport: FC = () => {
  const { data, status } = useGetUserProfile()

  useEffect(() => {
    if (status === "success") {
      loadCanny()
      Canny("identify", {
        appID: "5f0d32f03899af5d46779764",
        user: {
          // Replace these values with the current user's data
          email: data?.email,
          name: data?.name,
          id: data?.id,
          companies: [
            {
              name: data?.school.name,
              id: data?.school.id,
            },
          ],
        },
      })
    }
  }, [status, data])

  return (
    <Box mx="auto" sx={{ maxWidth: "maxWidth.sm" }}>
      <Typography.H6 mx={3} mt={4} mb={2}>
        <Trans>Help and Feedbacks</Trans>
      </Typography.H6>
      <Card sx={{ borderRadius: [0, "default"] }}>
        <a href="https://feedback.obserfy.com">
          <Flex p={3} sx={{ alignItems: "center" }}>
            <Box mr={3}>
              <Typography.Body>
                <Trans>Give Feedback</Trans>
              </Typography.Body>
              <Typography.Body color="textMediumEmphasis">
                <Trans>Post and upvote suggestions, ideas, and issues.</Trans>
              </Typography.Body>
            </Box>
            <Icon as={NextIcon} ml="auto" />
          </Flex>
        </a>

        <Flex p={3} sx={{ alignItems: "flex-start" }}>
          <Box mr={3}>
            <Typography.Body>
              <Trans>Shoot us an e-mail</Trans>
            </Typography.Body>
            <Typography.Body color="textMediumEmphasis">
              <Trans>
                Have a question? Shoot us an email at support@obserfy.com
              </Trans>
            </Typography.Body>
          </Box>
          <a
            href="mailto:support@obserfy.com"
            sx={{ ml: "auto", flexShrink: 0 }}
          >
            <Button variant="outline" px={2} py={1}>
              <Icon as={MailIcon} my={1} mx={0} ml="auto" />
            </Button>
          </a>
        </Flex>
      </Card>
    </Box>
  )
}

export default PageSupport
