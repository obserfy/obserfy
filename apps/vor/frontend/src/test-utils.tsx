/* eslint-disable import/no-extraneous-dependencies */
import { FC, ReactElement } from "react"
import { QueryClient, QueryClientProvider } from "react-query"
import { ThemeProvider, Box, Theme } from "theme-ui"
import { render, RenderOptions, RenderResult } from "@testing-library/react"
import {
  createHistory,
  createMemorySource,
  LocationProvider,
} from "@reach/router"
import { I18nProvider } from "@lingui/react"
import { i18n } from "@lingui/core"
import { en } from "make-plural/plurals"
import theme from "./gatsby-plugin-theme-ui"
// @ts-ignore
import enCatalog from "../i18n/lingui/en/messages.js"

// any used here because the catalogs are generated b lingui
// not our responsibility
i18n.loadLocaleData("en", { plurals: en })
i18n.load("en", enCatalog.messages as any)
i18n.activate("en")

const queryClient = new QueryClient()
const history = createHistory(createMemorySource("/"))

const AllProviders: FC = ({ children }) => {
  return (
    <I18nProvider i18n={i18n}>
      <QueryClientProvider client={queryClient}>
        <LocationProvider history={history}>
          <ThemeProvider theme={theme as Theme}>
            <Box sx={{ fontSize: [16, 20] }}>{children}</Box>
          </ThemeProvider>
        </LocationProvider>
      </QueryClientProvider>
    </I18nProvider>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult => render(ui, { wrapper: AllProviders, ...options })

// re-export everything
export * from "@testing-library/react"

// override render method
export { customRender as render }
