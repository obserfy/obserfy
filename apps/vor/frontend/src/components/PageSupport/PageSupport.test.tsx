import { render } from "../../test-utils"
import { Basic } from "./PageSupport.stories"

describe("PageSupport", () => {
  it("should render correctly", () => {
    const { container } = render(<Basic />)
    expect(container).toMatchSnapshot()
  })
})
