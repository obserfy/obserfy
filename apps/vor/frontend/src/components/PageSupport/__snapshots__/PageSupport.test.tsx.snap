// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageSupport should render correctly 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-size: 16px;
}

@media screen and (min-width: 40em) {
  .emotion-0 {
    font-size: 20px;
  }
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-left: auto;
  margin-right: auto;
  max-width: 640px;
}

.emotion-2 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  font-weight: bold;
  line-height: heading;
  color: var(--theme-ui-colors-text);
  text-rendering: optimizeLegibility;
  font-size: 20px;
  margin-left: 16px;
  margin-right: 16px;
  margin-top: 32px;
  margin-bottom: 8px;
}

.emotion-3 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 1px,rgba(0, 0, 0, 0.27) 0px 2px 4px -2px;
  border-radius: 0;
}

@media screen and (min-width: 40em) {
  .emotion-3 {
    border-radius: 6px;
  }
}

.emotion-4 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  margin-right: 16px;
}

.emotion-6 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
}

.emotion-7 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  color: var(--theme-ui-colors-text);
  font-size: 14px;
  text-rendering: optimizeLegibility;
  line-height: 1.5;
  color: var(--theme-ui-colors-textMediumEmphasis);
}

.emotion-8 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-left: auto;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.emotion-13 {
  margin-left: auto;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
}

.emotion-14 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  -ms-appearance: none;
  appearance: none;
  display: inline-block;
  text-align: center;
  line-height: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: inherit;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: var(--theme-ui-colors-primary);
  border: 0;
  border-radius: 4px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  cursor: pointer;
  box-sizing: border-box;
  font-size: 14px;
  color: var(--theme-ui-colors-textPrimary);
  background-color: var(--theme-ui-colors-surface);
  border-radius: 6px;
  font-family: Open Sans,-apple-system,BlinkMacSystemFont,San Francisco,Roboto,Segoe UI,Helvetica Neue,sans-serif;
  -webkit-transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  transition: background-color 125ms cubic-bezier(0.0, 0.0, 0.2, 1);
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 16px;
  padding-right: 16px;
  white-space: nowrap;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  touch-action: manipulation;
  box-shadow: none;
  border-color: var(--theme-ui-colors-border);
  border-style: solid;
  border-width: 1px;
  line-height: 1;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: 4px;
  padding-bottom: 4px;
}

.emotion-14:disabled {
  opacity: 0.38;
  cursor: auto;
}

.emotion-14:hover,
.emotion-14:focus {
  background-color: var(--theme-ui-colors-primaryLighter);
  border-color: var(--theme-ui-colors-primary);
}

.emotion-14:hover:disabled,
.emotion-14:focus:disabled {
  background-color: transparent;
  border-color: var(--theme-ui-colors-border);
}

.emotion-15 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  max-width: 100%;
  height: auto;
  margin-top: 4px;
  margin-bottom: 4px;
  margin-left: auto;
  margin-right: 0;
  fill: var(--theme-ui-colors-icon);
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: 20px;
  height: 20px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <h6
        class="emotion-2"
      >
        Help and Feedbacks
      </h6>
      <div
        class="emotion-3"
      >
        <a
          href="https://feedback.obserfy.com"
        >
          <div
            class="emotion-4"
          >
            <div
              class="emotion-5"
            >
              <p
                class="emotion-6"
              >
                Give Feedback
              </p>
              <p
                class="emotion-7"
              >
                Post and upvote suggestions, ideas, and issues.
              </p>
            </div>
            <div
              class="emotion-8"
            />
          </div>
        </a>
        <div
          class="emotion-9"
        >
          <div
            class="emotion-5"
          >
            <p
              class="emotion-6"
            >
              Shoot us an e-mail
            </p>
            <p
              class="emotion-7"
            >
              Have a question? Shoot us an email at support@obserfy.com
            </p>
          </div>
          <a
            class="emotion-13"
            href="mailto:support@obserfy.com"
          >
            <button
              class="emotion-14"
            >
              <div
                class="emotion-15"
              />
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
